spec/tests/workflow_maker_spec.rb

```diff
@@ -0,0 +1,172 @@
+#
+# (1) Redistributions of source code must retain the above copyright notice,
+# this list of conditions and the following disclaimer.
+#
+# (2) Redistributions in binary form must reproduce the above copyright notice,
+# this list of conditions and the following disclaimer in the documentation
+# and/or other materials provided with the distribution.
+#
+# (3) Neither the name of the copyright holder nor the names of any contributors
+# may be used to endorse or promote products derived from this software without
+# specific prior written permission from the respective party.
+#
+# (4) Other than as required in clauses (1) and (2), distributions in any form
+# of modifications or other derivative works may not use the "OpenStudio"
+# trademark, "OS", "os", or any other confusingly similar designation without
+# specific prior written permission from Alliance for Sustainable Energy, LLC.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE
+# UNITED STATES GOVERNMENT, OR THE UNITED STATES DEPARTMENT OF ENERGY, NOR ANY OF
+# THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+# OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *******************************************************************************
+require_relative './../spec_helper'
+
+RSpec.describe 'WorkFlow Maker' do
+  it 'should save annual results to xml file and verify them' do
+    file_name = 'building_151.xml'
+    xml_path = File.expand_path("./../files/#{file_name}", File.dirname(__FILE__))
+    expect(File.exist?(xml_path)).to be true
+
+    ns = 'auc'
+    doc = BuildingSync::Helper.read_xml_file_document(xml_path)
+    workflow_maker = BuildingSync::WorkflowMaker.new(doc, ns)
+    result = {}
+    result[:completed_status] = 'Success'
+
+    annual_results = {}
+    annual_results['total_site_energy_savings_mmbtu'] = 100
+    annual_results['total_source_energy_savings_mmbtu'] = 200
+    annual_results['total_energy_cost_savings'] = 300
+    annual_results['baseline_fuel_electricity_kbtu'] = 400
+    annual_results['fuel_electricity_kbtu'] = 500
+    annual_results['baseline_fuel_natural_gas_kbtu'] = 600
+    annual_results['fuel_natural_gas_kbtu'] = 700
+    annual_results['annual_peak_electric_demand_kw'] = 800
+
+    scenarios = workflow_maker.get_scenario_elements
+    scenarios.each do |scenario|
```

Comment:
Since savings / etc. will only be possible, we need to add in some checks / flags.  I feel like the workflow should be as follows:
1. check if baseline modeled scenario exists and has result.  if so, cache those results somewhere.
2. if baseline modeled results do exist, then savings / etc. for different packages of measures can be calculated.  else, we should only be calculating the value of non-savings values (i.e. we can still get what the annual or monthly values for the electricity and NG usage is)

additionally, lots of the logic fails when package_of_measures is nil::NilClass, so we need to either throw some errors or something.

Reply:
added baseline results to the xml file, too. https://github.com/BuildingSync/BuildingSync-gem/commit/5a1cdbcf946867059c6540497ad09f33799d5404

checking if package_of_measures is nil https://github.com/BuildingSync/BuildingSync-gem/commit/2b05f5ba1cbaf20a22512fbf996a0747bb19a232